src/main/java/com/google/api/codegen/util/SampleConfigSanitizer.java

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.yaml.snakeyaml.Yaml;
+
+public class SampleConfigSanitizer {
+
+  private SampleConfigSanitizer() {
+    // utility class
+  }
+
+  private static final String TYPE_FIELD_NAME = "type";
+  private static final String SCHEMA_VERSION_FIELD_NAME = "config_schema_version";
+  private static final String VALID_TYPE_VALUE =
+      "com.google.api.codegen.samplegen.v1p2.SampleConfigProto";
+  private static final String VALID_SCHEMA_VERSION = "1.2.0";
+
+  // Filter out sample configure files without proper type definition.
+  public static List<String> sanitize(List<String> sampleConfigFileNames) {
+    return sampleConfigFileNames
+        .stream()
+        .filter(SampleConfigSanitizer::isValidSampleConfig)
+        .collect(Collectors.toList());
+  }
+
+  @SuppressWarnings("unchecked")
+  private static boolean isValidSampleConfig(String sampleConfigFileName) {
+    try (Reader reader = new FileReader(sampleConfigFileName)) {
```

Comment:
A YAML file can have multiple YAML docs, so you should iterate over those.

Reply:
Added check for each doc in a file. If any of the docs in one file does not have valid `type` or `schema_version`. This is the best effort for now -- the `YamlReader` in api-compiler doesn't support multi-doc yaml files anyway.

(Honestly I don't think supporting multi-doc yamls is necessary, considering there are two top-level fields: `schema_version` and `type` whose values don't change that much. User can already configure any number of samples in one file)